lib/core/constants.js

```diff
@@ -104,25 +104,38 @@ const wellknownHeaderNames = /** @type {const} */ ([
 /** @type {Record<typeof wellknownHeaderNames[number]|Lowercase<typeof wellknownHeaderNames[number]>, string>} */
 const headerNameLowerCasedRecord = {}
 
+// Note: object prototypes should not be able to be referenced. e.g. `Object#hasOwnProperty`.
+Object.setPrototypeOf(headerNameLowerCasedRecord, null)
```

Comment:
What about `Object.create`:

```ts
const headerNameLowerCasedRecord = Object.create(null);
```

Or even just use `Map` directly:

```ts
const headerNameLowerCasedRecord = new Map();
```

Reply:
This line already existed - it's in the diff of the latest commit because I had originally put `wellknownHeaderNameBuffers` in between the declaration of `headerNameLowerCasedRecord` and the call to `Object.setProtoypeOf` but reordered it. In the pr's diff, that line isn't touched.

Reply:
Nice. But IMHO `Map` could still be considered. WDYT?

Reply:
Let's do that in a separate PR @SukkaW, would you like to send one?

Reply:
Some may think that Map is faster than Object. In fact, however, Object is faster in many cases when high frequency deletion, deletion of all entries, and ordered iterators are not required, and there is no difference in performance (assuming the key is a string and Object is a null prototype).

Reply:
> Some may think that Map is faster than Object. In fact, however, Object is faster in many cases when high frequency deletion, deletion of all entries, and ordered iterators are not required, and there is no difference in performance (assuming the key is a string and Object is a null prototype).

Indeed. And only true when not storing nullable value (otherwise `Map#has` would be more convenient).